lib/blueprinter/view_collection.rb

```diff
@@ -35,7 +35,9 @@ def fields_for(view_name)
     end
 
     def transformers(view_name)
-      views[view_name].transformers
+      included_transformers = gather_transformers_from_included_views(view_name)
+      all_transformers = included_transformers.concat(views[:default].view_transformers).uniq
+      all_transformers.empty? ? Blueprinter.configuration.default_transformers : all_transformers
```

Comment:
We need to make a distinction between the configured default transformers and transformers of the default view.

According to [this section](https://github.com/procore-oss/blueprinter#global-transforms) of the README, configured default transformers should only be used if there are no other transformers. That's fine, and I've moved that logic here.

But I don't believe transformers of the default view should be treated in the same way, which I've changed from how https://github.com/procore-oss/blueprinter/pull/363 handles it. Instead, there are merged with the other view transformers. This is the behavior I would expect, since fields on default view are merged with the view's other fields.

I've added tests for both of these situations.

Reply:
This is an edge case, but in what order should the transformers be applied? I would expect ones from the `default` view first, followed by included views, and finally any from the final view. Looks like this might apply the default view ones last. Thoughts?

Reply:
@njbbaer Currently, the default views transformer is not applied when calling with a specific view. Do we need to change this behavior?

Reply:
@jhollinger I prefer from last to first in views because that provides easy tracing of transformers for the end implementer. If we taking that route can we also follow the same for default too?

Reply:
@bhooshiek-narendiran I may not have defined my terms well enough. I was talking about "transformers from the default view", as opposed to "the default transformers configuration setting". All discussion of "default" below means "transformers from the default view".

Allow me to amend my comment. In the PR's current code, when transformers are present, they _appear_ to be applied in this order:

```
1) Transformers from intermediate views (top-down), 2) Transformers from the named view, 3) Transformers from :default view
```

That ordering strikes me as arbitrary, and I could see it leading to confusion in some edge cases. IMHO a much more predictable ordering would be "descending":

```
1) Transformers from :default view, 2) Transformers from intermediate views (top-down), 3) Transformers from the named view
```
@bhooshiek-narendiran suggested the reverse of that order, which I also think would be fine:

```
1) Transformers from the named view, 2) Transformers from intermediate views (bottom-up), 3) Transformers from :default view.
```

Reply:
Thanks @jhollinger, that's an important point. I'm inclined towards the "descending" order, which gives named views the opportunity to have the final say on their rendering, so to speak.

@bhooshiek-narendiran can you elaborate on why you prefer the reverse?

> Currently, the default views transformer is not applied when calling with a specific view. Do we need to change this behavior?

Yes, we want to change this behavior so that it's easier to apply transformers more broadly. When a field is added to the default view it is applied to all of the named views, likewise a transformer added to the default view should also be applied in addition to the named views.

Reply:
@njbbaer @jhollinger Apologize I also meant the same thing.
```
class Sample	
	transformer_view_default

	view_1 do
	transformer_1
	end

	view_2 do
	transformer_2
	end

	view_3 do 
	transformer_3
	end
end

```

Execution order of transformer =>` transformer_3,transformer_2,transformer_1 `
Do we need to merge **transformer_view_default** along with views transformer or only during views transformer not present.

But Adding it to first or last both seems to be a good addition.

Reply:
@jhollinger @bhooshiek-narendiran
I've made the PR changes described, so the execution order is now "top-down".

@bhooshiek-narendiran
Not quite. In your example (without `include` statements) each view would execute these transformers in this order:
`view_1` => `transformer_view_default`, `transformer_1`
`view_2` => `transformer_view_default`, `transformer_2`
`view_2` => `transformer_view_default`, `transformer_3`

Here's a more complete example, which uses included views:
```
class Sample
  transform TransformViewDefault
  
  view :view_1 do
    transform Transformer1
  end

  view :view_2 do
    transform Transformer2
    include :view_1
  end

  view :view_3 do
    transform Transformer3
    include :view_2
  end
end
```

`view_1` => `TransformViewDefault`, `Transformer1`
`view_2` => `TransformViewDefault`, `Transformer1`, `Transformer2`
`view_2` => `TransformViewDefault`, `Transformer1`, `Transformer2`, `Transformer3`

The idea is that the "more specific" transformers get executed later, so they can get the last word on their rendering.